src/coprocessor/dag/dag.rs

```diff
@@ -109,6 +112,81 @@ impl DAGContext {
         }
     }
 
+    pub fn handle_streaming_request(&mut self) -> Result<(Response, bool)> {
+        let mut record_cnt = 0;
+        let mut chunk = Chunk::new();
+        let mut start_key = None;
+        let mut remain = false;
+        loop {
+            if record_cnt >= self.stream_batch_row_limit {
+                remain = true;
+                break;
+            }
+            match self.exec.next() {
+                Ok(None) => break,
+                Ok(Some(row)) => {
+                    self.req_ctx.check_if_outdated()?;
+                    record_cnt += 1;
+                    if start_key.is_none() {
+                        start_key = self.exec.take_last_key().or_else(|| Some(vec![]));
```

Comment:
when would `or_else` happen?

Reply:
Some `Executor`s don't support `last_key`. In that cases, `take_last_key` will retrun `None`.

Reply:
Then why is `vec![]` safe for this case?